### PR TITLE
[chore] tidy testing environment

### PR DESCRIPTION
**Pull request by zkamvar**
I noticed that I was getting errors during testing because of a couple of factors unrelated to the actual validity of the code:

1. I did not have `testthis` installed
2. Folders created inside of `tempdir()` are not cleaned up leading to errors when re-running the tests because the folders are not empty.

I fixed the first with `skip_if_not_installed()` and the second by using `withr::local_tempdir()`

### CODE REVIEW

File: tests/testthat/test-check_tbl_colnames.R [hunk 1]
```diff
@@ -24,6 +24,7 @@ test_that("check_tbl_colnames validates correct files", {
 })
 
 test_that("check_tbl_colnames fails on files", {
+  skip_if_not_installed("testthis")
```
Comment (annakrystalli) [inline tests/testthat/test-check_tbl_colnames.R:27]:
```suggestion
```

File: tests/testthat/test-check_tbl_unique_round_id.R [hunk 2]
```diff
@@ -31,6 +31,7 @@ test_that("check_tbl_unique_round_id works", {
 
 
 test_that("check_tbl_unique_round_id fails correctly", {
+  skip_if_not_installed("testthis")
```
Comment (annakrystalli) [inline tests/testthat/test-check_tbl_unique_round_id.R:34]:
```suggestion
```

**Review by annakrystalli — CHANGES_REQUESTED** [review]
Thanks for the PR @zkamvar !

Fix for 2. I'm very happy to accept. Much cleaner indeed and makes working interactively with tests much smoother.

Regarding `testthis`, it is listed in `Suggests` and I would expect anyone contributing code to install the full list of dependencies and be able to run the full suite of tests, build all docs etc before making a PR. So I would prefer to remove those skips

**Comment by annakrystalli** [comment]
> Feel free to squash-merge this since it amounts to exactly one change effectively.
Honestly I'm not 100% percent how to that. Is it `git reset --soft HEAD~3`?

**Review by annakrystalli — APPROVED** [review]
All good!

**Comment by zkamvar** [comment]
> > Feel free to squash-merge this since it amounts to exactly one change effectively.
> 
> Honestly I'm not 100% percent how to do that. Is it `git reset --soft HEAD~3`?

screenshot showing the dropdown menu for "merge pull request" with three options for merging including "merge", "squash and merge", and "rebase and merge"

**Comment by annakrystalli** [comment]
> > > Feel free to squash-merge this since it amounts to exactly one change effectively.
> > 
> > 
> > Honestly I'm not 100% percent how to do that. Is it `git reset --soft HEAD~3`?
> 
> screenshot showing the dropdown menu for "merge pull request" with three options for merging including "merge", "squash and merge", and "rebase and merge"

Awesome! Thanks!

**Pull request merged**